File: api/internal/core/entity/workspace.go
package entity

import "time"

type Workspace struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	Icon      string    `json:"icon"`
	Color     string    `json:"color"`
	Position  int       `json:"position"`
	Collapsed bool      `json:"collapsed"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

File: website/src/data/mappers/workspace.ts
import type { ApiWorkspace } from '../contracts/workspace'
import type { Workspace } from '../../entities/workspace'

export function mapWorkspace(value: ApiWorkspace): Workspace {
  return {
    id: value.id,
    name: value.name,
    icon: value.icon,
    color: value.color,
    position: value.position,
    collapsed: value.collapsed,
    createdAt: value.createdAt,
    updatedAt: value.updatedAt,
  }
}
